include/fmt/format-inl.h

```diff
@@ -1337,7 +1337,7 @@ template <typename T> decimal_fp<T> to_decimal(T x) noexcept {
 
   if (r < deltai) {
     // Exclude the right endpoint if necessary.
-    if (r == 0 && z_mul.is_integer && !include_right_endpoint) {
+    if (r == 0 && (z_mul.is_integer & !include_right_endpoint)) {
```

Comment:
Why switch to bitwise AND? Better codegen?

Reply:
Yes. With logical AND, it generates three `test` instructions and three conditional branches. With bitwise AND, it generates two `test` instructions and two conditional branches. I thought compilers are aware of how to optimize this when all the things are just booleans variables (thus no short-circuit is really involved), but apparently they don't bother doing so. I am not sure why...

Reply:
I guess, compilers are not sure if all boolean variables involved are really just 0 or 1. Like, if one is `2` and another is `1`, they are both `true` so `&&` should give `true`, but `&` will give `0`. It is wrong from the first place to use `bool` for things with more than two states, but maybe the current standard allows `bool` variables to hold nonsensical values without UB.